src/sagemaker/automl/automlv2.py

```
@@ -0,0 +1,1283 @@
+        if wait:
+            self.latest_auto_ml_job.wait(logs=logs)
+
+    @classmethod
+    def attach(cls, auto_ml_job_name, sagemaker_session=None):
+        """Attach to an existing AutoML job.
+
+        Creates and returns a AutoML bound to an existing automl job.
+
+        Args:
+            auto_ml_job_name (str): AutoML job name
+            sagemaker_session (sagemaker.session.Session): A SageMaker Session
+                object, used for SageMaker interactions (default: None). If not
+                specified, the one originally associated with the ``AutoML`` instance is used.
+
+        Returns:
+            sagemaker.automl.AutoML: A ``AutoMLV2`` instance with the attached automl job.
+
+        """
+        sagemaker_session = sagemaker_session or Session()
+
+        auto_ml_job_desc = sagemaker_session.describe_auto_ml_job_v2(auto_ml_job_name)
+        automl_job_tags = sagemaker_session.sagemaker_client.list_tags(
+            ResourceArn=auto_ml_job_desc["AutoMLJobArn"]
+        )["Tags"]
+        inputs = [AutoMLDataChannel._from_api_response(channel) for channel in auto_ml_job_desc["AutoMLJobInputDataConfig"]]
+
+        problem_type = auto_ml_job_desc["AutoMLProblemTypeConfigName"]
+        problem_config = None
+        if problem_type == "ImageClassification":
+            problem_config = AutoMLImageClassificationConfig._from_api_response(auto_ml_job_desc["AutoMLProblemTypeConfig"])
+        elif problem_type == "TextClassification":
+            problem_config = AutoMLTextClassificationConfig._from_api_response(auto_ml_job_desc["AutoMLProblemTypeConfig"])
+        elif problem_type == "TimeSeriesForecasting":
+            problem_config = AutoMLTimeSeriesForecastingConfig._from_api_response(auto_ml_job_desc["AutoMLProblemTypeConfig"])
+        elif problem_type == "Tabular":
+            problem_config = AutoMLTabularConfig._from_api_response(auto_ml_job_desc["AutoMLProblemTypeConfig"])
+        elif problem_type == "TextGeneration":
```

should perhaps raise error on unknown problem type?

---

yeah, good point, I'll add the check here